### PR TITLE
Implement association relationship between classes

### PR DESCRIPTION
**Pull request by yasinabdmahmood**
### Project changes
- Created a class `Classroom` with the following:
  - `@label` instance variable, which is initialized in the constructor.
  - Setter and getter for `@label` (used  `attr_accessor`).
- Created the `has-many`/`belongs-to` relationship between `Classroom` and `Student`. The following is being implemented:
  - Created the `has-many` side (a classroom has many students).
  - Created the `belongs-to` side (a student belongs to a classroom).
  - Made sure that when adding a student to a classroom it also sets the classroom for the student.
  - Made sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Created a class `Book` with the following:
  - `@title` and `@author` instance variables, which are initialized in the constructor.
  - Setters and getters for instance variables (used `attr_accessor`).
- Created a class `Rental` with the following:
  - `@date` instance variable, which is initialized in the constructor.
  - Setter and getter for `@date` (used  `attr_accessor`).
- Created the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`. The following are implemented:
  - Created the `has-many` side of `Book` and `Rental` (a book has many rentals).
  - Created the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book).
  - Created the `has-many` side of `Person` and `Rental` (a person has many rentals).
  - Created the `belongs-to` side of `Rental` and `Person` (a rental belongs to a person).
  - Modified the constructor of `Rental` so `Book` and `Person` are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
```
Comment (t-yanick) [inline book.rb:11]: - [ ] To implement the has-many side of the relationship, a book has-many rentals, method for adding rentals should be present, something like so: 
```ruby
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
```
Please kindly fix that

File: person.rb [hunk 2]
```diff
@@ -2,14 +2,15 @@
...
+  attr_reader :id, :rentals
 
   def initialize(age, name = 'Unknown', parent_permission: true)
     super()
     @id = Random.rand(1...1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
```
Comment (t-yanick) [inline person.rb:16]: - [ ] To implement the has-many side of the relationship, a person has-many rentals, method for adding rentals should be present, something like so: 
```ruby
  def add_rental(book, date)
    Rental.new(date, self, book)
  end
```
Please kindly fix that

**Review by t-yanick — CHANGES_REQUESTED** [review]
Hi @yasinabdmahmoo, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N /A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Deepakdanger — APPROVED** [review]
Hi @yasinabdmahmood , 

### Status: Approved ✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚢 
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**